ci/jenkins/utils.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    echo "====== Checking installed Golang version ======"
+    antrea_golang_version="go$(grep -E "^go\ " go.mod | awk '{print $2}')"
+    golang_minor_version=$(echo ${antrea_golang_version} | awk -F '.' '{print $2}')
+
+    curl -s https://go.dev/dl/\?mode\=json\&include\=all | jq -c -r '.[]' | while read i; do
```

Comment:
I have run a test, and if the go.mod specifies `go 1.21.0`,  it tries to install 1.21.0 instead of 1.21.4, which doesn't seem right to me.

I am including a tab below, showing what I would expect to happen:
| go.mod version | Latest Patch Version for go.mod Minor Version | Latest Installed Version for go.mod Minor Version | Action |
| --------------- | --------------------------------------------- | ------------------------------------------------- | ------ |
| 1.21.0           | 1.21.4                                         | 1.21.4                                             | Use installed version (under `/usr/local/go-1.21`) |
| 1.21.0           | 1.21.4                                         | 1.21.4                                             | Install go 1.21.4 under `/usr/local/go-1.21` and use it |
| 1.21.0           | 1.21.4                                         | None                                             | Install go 1.21.4 under `/usr/local/go-1.21` and use it |
| 1.21           | 1.21.4                                         | 1.21.4                                             | Use installed version (under `/usr/local/go-1.21`) |
| 1.19           | 1.19.11                                         | 1.19.10                                             | Install go 1.19.11 under `/usr/local/go-1.19` and use it |

The first column value (`antrea_gomod_version`) is obtained with `grep -E "^go\ " go.mod | awk '{print $2}'`. 
The minor version (`antrea_gomod_minor_version`) is obtained with `echo "$antrea_gomod_version" | cut -d. -f1,2`, or equivalent `awk` command.
The second column value (`desired_version`) is obtained by querying https://go.dev/dl` and using `antrea_gomod_minor_version` to filter.
The third column value (`current_version`) can be obtained with `go version | awk '{print $3}' | cut -c 3-`.
You can then compare `desired_version` and `current_version`.

Reply:
Thanks for the table,  in the previous implementation I used the accucate golang version in `go.mod` if the version is in the format `go x.y.z`, now this script will always try to update to the latest patch release of Golang.